### PR TITLE
convert to SSG

### PR DESCRIPTION
**Pull request by royemosby**
# Enhancement - Convert Site Build to SSG

## Description

Converts site to use 11ty.

Please include a summary of the change and which issue is fixed.

- #31 
    - routing based through html template files + build process
- #29 
    - decomposes web page html into partials used by templates
- #54 
- #25
- #10 
- #5 
- #8 

## Type of change

- [ ] Bug fix
- [x] New feature
- [ ] Documentation
- [x] Infrastucture
- [ ] Other (please clarify): _____

## Change log

- Install, configure 11ty
- Decomposes existing html into re-usable partials
- Refactors styles, scripts to be compatible with build process
- Converts existing index file into template
- Creates index page
- Stubs out pages:
    - about
    - code of conduct
    - contact
    - events
    - media
    - 404
    - contribute (hidden in /drafts folder)
    - resources (hidden in /drafts folder)
    - team (hidden in /drafts folder)
- Adds placeholder directory for potential blog/announcements content
- pulls over existing favicons, assets, manifest

## Testing

- Visaual - deployed working branch to [Netlify as a staging environment](https://ndit-staging.netlify.app/)

## Checklist

- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new console warnings.
- [x] I manually tested to prove my fix is effective or that my feature works.
- [x] I have assigned this PR to an [owner](https://github.com/nditcommunity/ndit-website).

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,131 +1,5 @@
...
-.yarn/unplugged
-.yarn/build-state.yml
-.yarn/install-state.gz
-.pnp.*
-package-lock.json
+node_modules
+.DS_Store
+_site
+**/.DS_Store
+.DS_Store
```
Comment (marcusolsson) [inline .gitignore:5]: Seems like we have multiple entries for .DS_Store in the gitignore?

**Review by marcusolsson — COMMENTED** [review]
I think this is great as a base to build upon. First time hearing about Liquid though. What's the benefits over something like Nunjucks and Handlebars?

File: .gitignore (hunk 1 above)
Comment (royemosby) [reply .gitignore:5]: That's something that I'll clean up

**Comment by royemosby** [comment]
@marcusolsson 

> I think this is great as a base to build upon. First time hearing about Liquid though. What's the benefits over something like Nunjucks and Handlebars?

I don't think any one of the templating languages have much of an advantage over the other. This is one of those cases where the same type of tech was developed by different groups to satisfy their needs rather than coalescing around 1 tool.

My own use of Liquid is simply based on familiarity - it was the default templating for Jekyll when I started web dev and I have not had a reason to reach for anything else in JS. If you prefer another templating language, we can definitely go that way.

**Comment by marcusolsson** [comment]
It was the only one I hadn't heard of before so just curious. I don't mind it, especially if you're already familiar with it. As you say, they're all pretty similar. Just curious :)

**Pull request merged**